emop-maven-plugin/src/main/java/edu/cornell/AffectedSpecsMojo.java

```diff
@@ -67,32 +69,62 @@ private enum OutputFormat { BIN, TXT }
     public void execute() throws MojoExecutionException {
         super.execute();
         getLog().info("[eMOP] Invoking the AffectedSpecs Mojo...");
+        long start = System.currentTimeMillis();
+        // If only compute changed classes, then these lines can stay the same
         String[] arguments = createAJCArguments();
         Main compiler = new Main();
         MessageHandler mh = new MessageHandler();
         compiler.run(arguments, mh);
         IMessage[] ms = mh.getMessages(IMessage.WEAVEINFO, false);
+        long end = System.currentTimeMillis();
+        getLog().info("[eMOP Timer] Compile-time weaving takes " + (end - start) + " ms");
+        start = System.currentTimeMillis();
+        classToSpecs = readMapFromFile();
         computeMapFromMessage(ms);
-        writeMapToFile(classToSpecsFormat);
+        // TODO: Examine this idea
+//        if (changedMap.isEmpty()) {
+//            getLog().info("[eMOP] Changed classes are not associated with any specs, terminating...");
+//            System.exit(0);
+//        }
+        // Referenced from: https://stackoverflow.com/questions/7194522/how-to-putall-on-java-hashmap-contents-of-one-to-another-but-not-replace-existi
+        // Update map
+        changedMap.forEach((key, value) -> classToSpecs.merge(key, value, (oldValue, newValue) -> newValue));
+        computeAffectedSpecs();
+        end = System.currentTimeMillis();
+        getLog().info("[eMOP Timer] Compute affected specs takes " + (end - start) + " ms");
+        start = System.currentTimeMillis();
+        // Write map
+        writeMapToFile(OutputFormat.BIN);
+        // Write affectedSpecs
+        writeMapToFile(OutputFormat.TXT);
+        end = System.currentTimeMillis();
+        getLog().info("[eMOP Timer] Write affected specs to disk takes " + (end - start) + " ms");
         getLog().info("[eMOP] Number of impacted classes: " + getImpacted().size());
         getLog().info("[eMOP] Number of messages to process: " + Arrays.asList(ms).size());
     }
 
+    private void computeAffectedSpecs() throws MojoExecutionException {
+        for (String impactedClass : getImpacted()) {
+            Set<String> associatedSpecs = classToSpecs.get(impactedClass);
+            if (associatedSpecs != null) {
```

Comment:
what if the class had no associated specs before but now has some associated specs? (i.e. if `associatedSpecs == null`?)